src/toxcore/net_crypto/crypto_connection.rs

```diff
@@ -433,9 +464,60 @@ impl CryptoConnection {
         }
     }
 
-    /// Set time when last UDP packet was received to now
-    pub fn update_udp_received_time(&mut self) {
-        self.udp_received_time = Some(clock_now())
+    /// Set UPD address for this connection
+    pub fn set_udp_addr(&mut self, addr: SocketAddr) {
+        match addr {
+            SocketAddr::V4(addr) => self.udp_addr_v4 = Some(ConnectionAddr::new(addr)),
```

Comment:
https://doc.rust-lang.org/std/option/enum.Option.html#method.replace

Reply:
Replace is used to get old value which I don't need. I just need to put a new value there.

Reply:
Too bad Option does not have a method like `::set`.

Reply:
I think it's because `=` does exact same thing :)

Reply:
:D